src/output.F90

```diff
@@ -2162,5 +2184,239 @@ function get_label(t, i_filter) result(label)
+        if (c % type /= CELL_NORMAL) then
+
+          ! There are more cells in this universe that it could be in
+          later_cell = .true.
+
+          ! Two cases, lattice or fill cell
+          if (c % type == CELL_FILL) then
+            temp_offset = c % offset(map)
+
+          ! Get the offset of the first lattice location
+          else
+            lat => lattices(c % fill) % obj
+            temp_offset = lat % offset(map, 1, 1, 1)
+          end if   
+
+          ! If the final offset is in the range of offset - temp_offset+offset
+          ! then the goal is in this cell
+          if (final < temp_offset + offset) then
+            this_cell = .true.
+          end if  
+        end if
+      end if
+
+      if (n == 1 .and. c % type /= 1) then
+        this_cell = .true.
+      end if
+
+      if (.not. later_cell) then
+        this_cell = .true.
+      end if
+
+      ! Get pointer to THIS cell because target must be in this cell
+      if (this_cell) then
+
+        cell_index = univ % cells(i)
+        c => cells(cell_index)
+
+        path = trim(path) // "->" // to_str(c%id)
+
+        ! ====================================================================
+        ! CELL CONTAINS LOWER UNIVERSE, RECURSIVELY FIND CELL
+        if (c % type == CELL_FILL) then
+
+          ! Enter this cell to update the current offset
+          offset = c % offset(map) + offset
+
+          next_univ => universes(c % fill)
+          call find_offset(map, goal, next_univ, final, offset, path)
+          return
+
+        ! ====================================================================
+        ! CELL CONTAINS LATTICE, RECURSIVELY FIND CELL
+        elseif (c % type == CELL_LATTICE) then
+
+          ! Set current lattice
+          lat => lattices(c % fill) % obj
+
+          select type (lat)
+
+          ! ==================================================================
+          ! RECTANGULAR LATTICES
+          type is (RectLattice)
+
+            ! Write to the geometry stack
+            path = trim(path) // "->" // to_str(lat%id)
+
+            n_x = lat % n_cells(1)
+            n_y = lat % n_cells(2)
+            n_z = lat % n_cells(3)
+
+            ! Loop over lattice coordinates
+            do m = 1, n_z
+              do l = 1, n_y
+                do k = 1, n_x
+
+                  if (final > lat % offset(map, k, l, m) + offset) then
+                    old_m = m
+                    old_l = l
+                    old_k = k 
+                   cycle
+                  else
+                    ! Target is at this lattice position
+                    lat_offset = lat % offset(map, k, l, m)
+                    offset = offset + lat_offset
+                    next_univ => universes(lat % universes(k, l, m))  
+                    path = trim(path) // "(" // trim(to_str(k)) // &
+                         "," // trim(to_str(l)) // "," // &
+                         trim(to_str(m)) // ")"
```

Comment:
Actually, this whole block for ``RectLattice`` needed some fixing, which I believe I accomplished with my latest commit. To be perfectly honest, I never use "tallies.out" and only process tallies with the statepoint files. So I haven't fully vetted the output of distribcell path strings for complicated geometries. I've run some cases with both simple and nested lattices (both rectangular and hexagonal) and things are everything now appears as I expect it should.